scipy/signal/ltisys.py

```diff
@@ -121,10 +121,10 @@ def tf2ss(num, den):
     if num.shape[-1] > 0:
         D = num[:, 0]
     else:
-        D = array([], float)
+        D = array([0], float)
```

Comment:
Yeah...I should probably comment on that in the codebase.  I actually disagree that `array([0])` is equivalent to `array()`.  Otherwise #5760 should never have been an issue. :smile:

As I noted in my PR overview, `array([0])` should be used to differentiate from returning a `null` array, which is what you get when the orders of `num` and `den` are zero.  These arrays have no reason to be processed by `ss2tf` or `tf2ss`.  On the other hand, "empty" arrays that are needed in specifying the state-space equation need to have dimensions and not be empty in the same sense as a `null` array so that they can be handled later on down the road.